### PR TITLE
Switch to using sam build for building python Lambdas

### PR DESCRIPTION
**Pull request by abrooksv**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
Refactor LambdaPackager to expose a build from template method. If template is defined in run config, use that as input, else make one on the fly. 

Java goes the opposite way and extracts what it needs from a template until we migrate that to sam build.

Output template is then used as input to sam local invoke.

## Related Issue(s)
#483 

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by abrooksv** [comment]
* Can now request to use containers to build. This fixes our tests for building.
* Path mapping has been disabled in tests since it all needs to change to support sam build artifacts.
* Python run config tests are disabled due to they need containers to run but that is not exposed yet as an option

**Comment by abrooksv** [comment]
Path mappings have been re-enabled and tests updated for them

**Comment by abrooksv** [comment]
Ready for review, test gaps have been closed

**Comment by abrooksv** [comment]
Integration tests passed with latest commit: aws-jetbrains-toolkit-integ-test:e81ef82d-6515-40c2-a79c-6d67fa2880a8

File: jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/deploy/SamDeployTest.kt [hunk 1]
```diff
@@ -154,7 +154,7 @@ class SamDeployTest {
                     parameters,
                     bucketRule.createBucket(stackName),
                     false,
-                    false
+                    true
```
Comment (rli) [inline jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/deploy/SamDeployTest.kt]: add a test for non-containerized build?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            val pathMappings = listOf(
+                PathMapping(templateLocation.parent.resolve(codeLocation).toString(), "/"),
+                PathMapping(buildDir.resolve(logicalId).toString(), "/")
+            )
+
+            val processHandler = ProcessHandlerFactory.getInstance().createColoredProcessHandler(commandLine)
+            processHandler.addProcessListener(object : ProcessAdapter() {
+                override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
+                    // TODO: We should find a way to show the output of this in the UI
+                    println(event.text)
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt]: Ill log for now, follow up PR to get it into the UI

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/PathMapper.kt [hunk 3]
```diff
@@ -49,4 +50,18 @@ class PathMapping(localPath: String, remotePath: String) {
     internal val remoteRoot = FileUtil.normalize("$remotePath/")
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/PathMapper.kt]: I could not figure out a way to insert the normalize call, if you know how please let me know

File: jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/deploy/SamDeployTest.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/deploy/SamDeployTest.kt]: Sure, with sam 0.12, thats easy

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                    logicalId,
+                    templateLocation
+                )
+            )
+
+        ApplicationManager.getApplication().executeOnPooledThread {
+            val buildDir = FileUtil.createTempDirectory("lambdaBuild", null, true).toPath()
+
+            val commandLine = SamCommon.getSamCommandLine()
+                .withParentEnvironmentType(GeneralCommandLine.ParentEnvironmentType.CONSOLE)
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt]: Done from all 3 locations

**Pull request merged**